DeDriftAndResample/scripts/ExtractFromMRFIXConcat.sh

```diff
@@ -0,0 +1,148 @@
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder containing all subjects"
+opts_AddMandatory '--subject' 'Subject' 'ID' "the subject ID"
+opts_AddMandatory '--multirun-fix-names' 'mrfixNames' 'run1@run2...' "list of run names used in MR FIX, in the SAME ORDER, separated by @s"
+opts_AddOptional '--multirun-fix-names-to-use' 'mrfixNamesToUse' 'rest1@rest2...' "list of runs to extract (for cifti or volume outputs)"
+opts_AddOptional '--concat-cifti-input' 'concatCifti' 'file' "filename of the concatenated cifti (for making cifti output)"
+opts_AddOptional '--cifti-out' 'ciftiOut' 'file' "output filename for extracted cifti data"
+opts_AddOptional '--concat-volume-input' 'concatVol' 'file' "filename of the concatenated volume (for making volume output)"
+opts_AddOptional '--volume-out' 'volOut' 'file' "output filename for extracted volume data"
+opts_AddOptional '--csv-out' 'csvOut' 'file' "output filename for csv of start and end index for all run names"
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+if [[ -z "$ciftiOut" && -z "$volOut" && -z "$csvOut" ]]
+then
+    log_Err_Abort "no outputs options were specified (use one or more of --cifti-out, --volume-out, or --csv-out)"
+fi
+
+if [[ -n "$ciftiOut" && (-z "$concatCifti" || -z "$mrfixNamesToUse") ]]
+then
+    log_Err_Abort "using --cifti-out requires specifying --concat-cifti-input and --multirun-fix-names-to-use"
+fi
+
+if [[ -n "$volOut" && (-z "$concatVol" || -z "$mrfixNamesToUse") ]]
+then
+    log_Err_Abort "using --volume-out requires specifying --concat-volume-input and --multirun-fix-names-to-use"
+fi
+
+doMerge=0
+if [[ -n "$ciftiOut" || -n "$volOut" ]]
+then
+    doMerge=1
+    IFS=@ read -a mrNamesUseArray <<< "$mrfixNamesToUse"
+fi
+
+IFS=@ read -a mrNamesArray <<< "$mrfixNames"
+#sanity check for identical names
+for ((index = 0; index < ${#mrNamesArray[@]}; ++index))
+do
+    for ((index2 = 0; index2 < ${#mrNamesArray[@]}; ++index2))
+    do
+        if ((index != index2)) && [[ "${mrNamesArray[$index]}" == "${mrNamesArray[$index2]}" ]]
+        then
+            log_Err_Abort "MR fix names list contains '${mrNamesArray[$index]}' more than once"
+        fi
+    done
+done
+
+if [[ -n "$csvOut" ]]
+then
+    rm -f -- "$csvOut"
+fi
+
+runSplits=()
+runIndices=()
+curTimepoints=0
+#convention: one before the first (1-based) index of the run
+runSplits[0]="$curTimepoints"
+#calculate the timepoints where the concatenated switches runs, find which runs are used
+for ((index = 0; index < ${#mrNamesArray[@]}; ++index))
+do
+    fmriName="${mrNamesArray[$index]}"
+    NumTPS=$(wb_command -file-information "$StudyFolder/$Subject/MNINonLinear/Results/$fmriName/${fmriName}_Atlas.dtseries.nii" -only-number-of-maps)
```

Comment:
Note that this currently hardcodes to look at the MSMSulc uncleaned per-run cifti files to get the length of each run.

Reply:
If this should be improved, I'll need to know which files are likely to be available in the scenarios we want to support.

Reply:
Volume:
$StudyFolder/$Subject/MNINonLinear/Results/$fmriName/${fmriName}.nii.gz

RegName="NONE":
$StudyFolder/$Subject/MNINonLinear/Results/$fmriName/${fmriName}_Atlas.dtseries.nii

RegName="MSMAll" or RegName="${RegName}"
$StudyFolder/$Subject/MNINonLinear/Results/$fmriName/${fmriName}_Atlas_MSMAll.dtseries.nii
$StudyFolder/$Subject/MNINonLinear/Results/$fmriName/${fmriName}_Atlas_${RegName}.dtseries.nii

Reply:
Is it possible someone will have a concatenated volume file but not those per-run volume files?

Reply:
No.

Reply:
So, what's the plan here?  Just always get the length of each run from the MSMSulc uncleaned per-run cifti files (as currently implemented)?  I don't have a strong opinion either way.  While it seems like it might be relatively easy to add a `RegName` argument to select between MSMSulc and MSMAll, if we go the route of tailoring to `RegName`, then we should probably tailor to the volume files as well, which would require adding a volume-specific section to retrieving `NumTPSVol`.

Reply:
Ideally we would pull the info from the type of data we are splitting.

Reply:
This now takes an optional regname for the cifti to test for run length, and if the only data output requested is volume, takes the length from volume files.  At present, the regname should be "" for MSMSulc when calling Extract..., as it doesn't have the internal naming convention for that, DeDrift... handles that instead.

Reply:
Extract... should now accept `MSMSulc`.